packages/trader/src/Modules/SmartChart/Components/AllMarkers/Helpers/svg2canvas.js

```diff
@@ -0,0 +1,87 @@
+import { calc_scale } from './calculations';
+
+const Svg2Canvas = (() => {
+    /** @param {CanvasRenderingContext2D} ctx */
+    const render = ({ ctx, icon, position: { zoom, top, left } }) => {
+        ctx.save();
+        const scale = calc_scale(zoom);
+
+        ctx.translate(left - (icon.width * scale) / 2, top - (icon.height * scale) / 2);
+
+        ctx.scale(scale, scale);
+
+        icon.paths.forEach(({ points, fill, stroke }) => {
+            if (fill) {
+                ctx.fillStyle = fill;
+            }
+            if (stroke) {
+                ctx.strokeStyle = stroke;
+            }
+            ctx.beginPath();
+            let prev_x, prev_y;
+            for (let idx = 0; idx < points.length; idx++) {
+                let x, y, cx1, cx2, cy1, cy2, r;
+                if (points[idx] === 'M') {
+                    x = points[++idx];
+                    y = points[++idx];
+                    ctx.moveTo(x, y);
+                } else if (points[idx] === 'L') {
+                    x = points[++idx];
+                    y = points[++idx];
+                    ctx.lineTo(x, y);
+                } else if (points[idx] === 'V') {
+                    y = points[++idx];
+                    ctx.lineTo(prev_x, y);
+                } else if (points[idx] === 'H') {
+                    x = points[++idx];
+                    ctx.lineTo(x, prev_y);
+                } else if (points[idx] === 'Q') {
+                    cx1 = points[++idx];
+                    cy1 = points[++idx];
+                    x = points[++idx];
+                    y = points[++idx];
+                    ctx.quadraticCurveTo(cx1, cy1, x, y);
+                } else if (points[idx] === 'C') {
+                    cx1 = points[++idx];
+                    cy1 = points[++idx];
+                    cx2 = points[++idx];
+                    cy2 = points[++idx];
+                    x = points[++idx];
+                    y = points[++idx];
+                    ctx.bezierCurveTo(cx1, cy1, cx2, cy2, x, y);
+                } else if (points[idx] === 'A') {
+                    x = points[++idx];
+                    y = points[++idx];
+                    r = points[++idx];
+                    const start_a = points[++idx];
+                    const end_a = points[++idx];
+                    ctx.arc(x, y, r, start_a, end_a);
+                }
+                prev_x = x;
+                prev_y = y;
+            }
+            ctx.closePath();
+            if (fill) {
+                ctx.fill();
+            }
+            if (stroke) {
+                ctx.stroke();
+            }
+        });
+        ctx.scale(1, 1);
+        ctx.restore();
+    };
+
+    const render_from_url = async ({ url }) => {
+        const response = await this.fetch(url);
+        const svg_xml = await response.text();
+        return render(svg_xml);
+    };
+
+    return {
+        render,
+        render_from_url,
+    };
+})();
```

Comment:
is there a reason to immediately invoked this function?

Reply:
@kelly-binary so that we it returns the functions in them. This is a bit incomplete, we could just remove all of these and use a function but it will be done in another card.